metricflow-semantics/metricflow_semantics/model/semantics/linkable_spec_resolver.py

```diff
@@ -244,25 +217,34 @@ def _get_semantic_model_for_measure(self, measure_reference: MeasureReference) -
             )
         return semantic_models_where_measure_was_found[0]
 
-    def _get_joinable_metrics_for_semantic_model(self, semantic_model: SemanticModel) -> LinkableElementSet:
-        linkable_metrics = []
-        for metric_ref in self._joinable_metrics_for_semantic_models.get(semantic_model.reference, set()):
-            for entity_link in [entity.reference for entity in semantic_model.entities]:
-                linkable_metrics.append(
-                    LinkableMetric(
-                        properties=frozenset({LinkableElementProperty.METRIC, LinkableElementProperty.JOINED}),
-                        join_path=MetricSubqueryJoinPath(
-                            metric_subquery_join_path_element=MetricSubqueryJoinPathElement(
-                                metric_reference=metric_ref, join_on_entity=entity_link
-                            )
+    def get_joinable_metrics_for_semantic_model(
+        self, semantic_model: SemanticModel, using_join_path: Optional[SemanticModelJoinPath] = None
+    ) -> LinkableElementSet:
+        """Get the set of linkable metrics that can be joined to this semantic model."""
+        properties = frozenset({LinkableElementProperty.METRIC, LinkableElementProperty.JOINED})
+        if using_join_path:
+            assert (
+                semantic_model.reference == using_join_path.last_semantic_model_reference
+            ), "Last join path element should match semantic model when building LinkableMetrics."
+            properties = properties.union(frozenset({LinkableElementProperty.MULTI_HOP}))
+
+        path_key_to_linkable_metrics: Dict[ElementPathKey, Tuple[LinkableMetric, ...]] = {}
+        for entity_reference in [entity.reference for entity in semantic_model.entities]:
+            if using_join_path and entity_reference in using_join_path.entity_links:
+                continue
+            for metric_reference in self._joinable_metrics_for_entities[entity_reference]:
+                linkable_metric = LinkableMetric(
+                    properties=properties,
+                    join_path=MetricSubqueryJoinPath(
+                        metric_subquery_join_path_element=MetricSubqueryJoinPathElement(
+                            metric_reference=metric_reference, join_on_entity=entity_reference
                         ),
-                    )
+                        semantic_model_join_path=using_join_path,
+                    ),
                 )
-        return LinkableElementSet(
-            path_key_to_linkable_metrics={
-                linkable_metric.path_key: (linkable_metric,) for linkable_metric in linkable_metrics
-            },
-        )
+                path_key_to_linkable_metrics[linkable_metric.path_key] = (linkable_metric,)
```

Comment:
This is overwriting rather than extending. Do we want to verify that they're all the same `linkable_metric`?

Reply:
Ok yeah I fixed this in a later PR (why is this stack such a mess) but I can add a fix here. Notable though that that is the same behavior for all linkable elements in functions where we build `LinkableDimensions` and `LinkableEntities`, so I can fix this for those too.

Reply:
Updated!